dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/QuayImageRegistry.java

```diff
@@ -133,11 +155,65 @@ public QuayImageRegistry(final Token quayToken) {
+        LOG.info(quayToken.getUsername() + " ======================= Getting image for tag {}================================", quayTag.getName());
+        QuayRepoManifest quayRepoManifest;
+        DockerManifestList manifestList;
+        List<Image> images = new ArrayList<>();
+        quayRepoManifest = manifestApi.getRepoManifest(quayTag.getManifestDigest(), tool.getNamespace() + '/' + tool.getName());
+        try {
+            manifestList = g.fromJson(quayRepoManifest.getManifestData(), DockerManifestList.class);
+        } catch (JsonSyntaxException ex) {
+            LOG.info("Unexpected response from Quay while retrieving repo manifest for Quay Tag " + quayTag.getName() + " for tool " + tool.getToolPath(), ex);
+            return images;
+        }
+        try {
+            Arrays.stream(manifestList.getManifests()).forEach(manifest -> {
+                Optional<QuayTag> matchingTag = cleanedQuayTagList.stream().filter(tag -> (tag.getManifestDigest().equals(manifest.getDigest()))).findFirst();
+                if (matchingTag.isPresent()) {
+                    final String manifestDigest = manifest.getDigest();
+                    final String imageID = matchingTag.get().getImageId();
+                    List<Checksum> checksums = new ArrayList<>();
+                    String[] splitChecksum = manifestDigest.split(":");
+                    checksums.add(new Checksum(splitChecksum[0], splitChecksum[1]));
+                    Image image = new Image(checksums, tool.getNamespace() + '/' + tool.getName(), quayTag.getName(), imageID, Registry.QUAY_IO, quayTag.getSize(), quayTag.getLastModified());
+                    image.setArchitecture(manifest.getPlatform().getArchitecture());
+                    image.setOs(manifest.getPlatform().getOs());
+                    images.add(image);
+                    cleanedQuayTagList.remove(matchingTag.get());
+                } else {
+                    LOG.info("Unable to find matching Quay tag image information using digest");
+                }
+            });
+        } catch (NullPointerException ex) {
+            LOG.info("Unexpected null response from Quay while retrieving image and checksum info from manifest for Quay Tag " + quayTag.getName() + " for tool " + tool.getToolPath(), ex);
+        }
+        return images;
+    }
+
+    private Tag convertQuayTagToTag(QuayTag quayTag, Tool tool, Map<QuayTag, List<Image>> multiImageQuayTags) throws InvocationTargetException, IllegalAccessException {
         final Tag tag = new Tag();
         BeanUtils.copyProperties(tag, quayTag);
-        Optional<Image> tagImage = getImageForTag(tool, tag, quayTag);
-        tagImage.ifPresent(image -> tag.getImages().add(image));
+        // If we were unable to get the list of images from the manifest list, then an empty image list will be added to the tag.
+        if (quayTag.isIsManifestList() != null && quayTag.isIsManifestList()) {
```

Comment:
Might use `Boolean.TRUE.equals(x)` here as you do elsewhere, it's probably a bit more understandable and less repetitive.